sormas-ui/src/main/java/de/symeda/sormas/ui/events/eventParticipantsLineListing/layout/LineListingLayout.java

```diff
@@ -0,0 +1,360 @@
+import com.vaadin.ui.VerticalLayout;
+import com.vaadin.ui.Window;
+import com.vaadin.ui.themes.ValoTheme;
+
+import de.symeda.sormas.api.FacadeProvider;
+import de.symeda.sormas.api.caze.CaseDataDto;
+import de.symeda.sormas.api.event.EventDto;
+import de.symeda.sormas.api.event.EventParticipantDto;
+import de.symeda.sormas.api.i18n.Captions;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.i18n.Strings;
+import de.symeda.sormas.api.infrastructure.district.DistrictReferenceDto;
+import de.symeda.sormas.api.infrastructure.region.RegionReferenceDto;
+import de.symeda.sormas.api.person.PersonDto;
+import de.symeda.sormas.api.utils.ValidationRuntimeException;
+import de.symeda.sormas.ui.UserProvider;
+import de.symeda.sormas.ui.utils.ButtonHelper;
+import de.symeda.sormas.ui.utils.CssStyles;
+import de.symeda.sormas.ui.utils.FieldHelper;
+import de.symeda.sormas.ui.utils.components.linelisting.line.DeleteLineEvent;
+import de.symeda.sormas.ui.utils.components.linelisting.line.LineLayout;
+import de.symeda.sormas.ui.utils.components.linelisting.model.LineDto;
+import de.symeda.sormas.ui.utils.components.linelisting.person.PersonField;
+import de.symeda.sormas.ui.utils.components.linelisting.person.PersonFieldDto;
+import de.symeda.sormas.ui.utils.components.linelisting.section.LineListingSection;
+
+public class LineListingLayout extends VerticalLayout {
+
+	private final ComboBox<RegionReferenceDto> region;
+	private final ComboBox<DistrictReferenceDto> district;
+
+	private final EventDto eventDto;
+	private final List<EventParticipantLineLayout> eventParticipantLines;
+	private final Window window;
+	private Consumer<LinkedList<LineDto<EventParticipantDto>>> saveCallback;
+
+	public LineListingLayout(Window window, EventDto eventDto) {
+		this.window = window;
+
+		setSpacing(false);
+
+		LineListingSection sharedInformationComponent = new LineListingSection(Captions.lineListingSharedInformation);
+
+		this.eventDto = eventDto;
+
+		HorizontalLayout sharedInformationBar = new HorizontalLayout();
+		sharedInformationBar.addStyleName(CssStyles.SPACING_SMALL);
+
+		region = new ComboBox<>(I18nProperties.getPrefixCaption(CaseDataDto.I18N_PREFIX, CaseDataDto.RESPONSIBLE_REGION));
+		region.setItemCaptionGenerator(item -> item.buildCaption());
+		region.setId("lineListingRegion");
+		sharedInformationBar.addComponent(region);
+
+		region.addValueChangeListener(e -> {
+			RegionReferenceDto regionDto = e.getValue();
+			updateDistricts(regionDto);
+			getEventParticipantLineDtos().forEach(eventParticipantDto -> {
+				eventParticipantDto.getEntity().setRegion(regionDto);
+			});
+		});
+
+		district = new ComboBox<>(I18nProperties.getPrefixCaption(CaseDataDto.I18N_PREFIX, CaseDataDto.RESPONSIBLE_DISTRICT));
+		district.setItemCaptionGenerator(item -> item.buildCaption());
+		district.setId("lineListingDistrict");
+		district.addValueChangeListener(e -> {
+			getEventParticipantLineDtos().forEach(eventParticipantDto -> {
```

Comment:
direct updates to the dtos must be avoided

Reply:
changed the approach